content/guides/styling-your-components.mdx

```diff
@@ -0,0 +1,522 @@
+
+## CSS modules
+
+CSS Modules are CSS files where class names, animations, and media queries are scoped locally by default. 
+These provide a way to encapsulate styles within components, preventing global conflicts and optimizing the final output by bundling only the used selectors.
+
+### Creating CSS module files
+
+Begin by creating a CSS module file. 
+Conventionally, these files have a `.module.css` extension, like `style.module.css`. 
+However, other other extensions can also be used, such as `.scss` and `.sass`.
+
+```css
+/* styles.module.css */
+.foo {
+  color: red;
+}
+.bar {
+  background-color: blue;
+}
+``` 
+**Note:** Avoid the use of HTML tags in CSS modules, as they are not considered pure selectors.
+
+### Using modules in components
+
+1. **Importing styles:** In your component file (eg. `Component.jsx`), import the styles from the CSS module.
+
+```jsx
+// component.jsx
+import styles from "styles.module.css";
+```
+
+2. **Applying styles:** Use the imported styles by referencing them as properties of the styles object in your JSX:
+
+```jsx
+function Component() {
+  return (
+    <>
+      <div class={`${styles.foo} ${styles.bar}`} >
+        Hello, world!
+      </div>
+    </>
+  );
+}
+```
+
+3. **Using a single style:** If you only need one style from the module, import and apply it directly:
+
+```jsx
+// component.jsx
+import styles from "styles.module.css";
+
+function Component() {
+  return (
+    <>
+      <div class={styles.foo}>
+        Hello, world!
+      </div>
+    </>
+  );
+}
+```
+
+4. **Mixing with regular class names:** CSS module syntax can be combined with regular string class names, as well:
+
+```jsx
+// component.jsx
+import styles from "styles.module.css";
+
+function Component() {
+  return (
+    <>
+      <div class={`${styles.foo} container`}>
+        Hello, world!
+      </div>
+    </>
+  );
+}
+```
+
+**Note:** If your styles have dashes in their names, use bracket notation:
+
+```jsx
+const className = styles["foo-with-dash"];
+```
+
+## CSS-in-JS
+
+CSS-in-JS is a modern approach to styling components.
+Within the [Solid ecosystem](/ecosystem), there have been various libraries and solutions available for working with CSS-in-JS, including but not limited to:
+
+* [Solid Styled Components](https://github.com/solidjs/solid-styled-components)
+* [Solid Styled JSX](https://github.com/solidjs/solid-styled-jsx)
+
+CSS-in-JS libraries often come with their own set of APIs and methods for defining, updating, and applying styles dynamically.
+Many also offer features like theming, media queries, and server-side rendering support right out of the box.
+
+**Note:** Before choosing a CSS-in-JS library, it is recommended to check its compatability with Solid.
+
+### Macaron
+
+[Macaron](https://macaron.js.org/) is compile-time CSS-in-JS library that offers type safety. 
+
+#### Installation
+
+1. Install and setup the Macaron plugin for your bundler:
```

Comment:
```suggestion
1. Install and set up the macaron plugin for your bundler:
```

Reply:
Note: "setup" is the noun, "to set up" is the verb.

Also, I checked. The developers don't appear to capitalize "macaron" outside of it appearing in the first word of a sentence. Annoying.